bin/plugin/commands/performance.js

```diff
@@ -187,226 +218,248 @@ function curateResults( testSuite, results ) {
- * @param {string} runKey                   Unique identifier for the test run, e.g. `branch-name_post-editor_run-3`.
- *
- * @return {Promise<WPPerformanceResults>} Performance results for the branch.
+ * @param {string} testSuite Name of the tests set.
+ * @param {string} runKey    Unique identifier for the test run, e.g. 'post-editor_ref-name_run-3'.
+ * @return {Promise<WPPerformanceResults>} Performance results for the ref.
  */
-async function runTestSuite( testSuite, performanceTestDirectory, runKey ) {
+async function runTestSuite( testSuite, runKey ) {
 	const resultsFilename = `${ runKey }.performance-results.json`;
 
-	await runShellScript(
-		`npm run test:performance -- ${ testSuite }`,
-		performanceTestDirectory,
-		{
-			...process.env,
-			WP_ARTIFACTS_PATH: ARTIFACTS_PATH,
-			RESULTS_FILENAME: resultsFilename,
-		}
-	);
+	await runShellScript( `npm run test:performance -- ${ testSuite }`, CWD, {
+		...process.env,
+		WP_ARTIFACTS_PATH: ARTIFACTS_PATH,
+		RESULTS_FILENAME: resultsFilename,
+	} );
 
 	return curateResults(
 		testSuite,
-		await readJSONFile( path.join( ARTIFACTS_PATH, resultsFilename ) )
+		readJSONFile( path.join( ARTIFACTS_PATH, resultsFilename ) )
 	);
 }
 
 /**
- * Runs the performances tests on an array of branches and output the result.
+ * Runs the performances tests on an array of git refs and output the result.
+ * Ref can be a branch, commit SHA or a tag.
  *
- * @param {string[]}                    branches Branches to compare
- * @param {WPPerformanceCommandOptions} options  Command options.
+ * @param {string[]}                    refs    Refs to compare.
+ * @param {WPPerformanceCommandOptions} options Command options.
  */
-async function runPerformanceTests( branches, options ) {
+async function runPerformanceTests( refs, options ) {
+	console.time( TIMERS.total );
+
 	const runningInCI = !! process.env.CI || !! options.ci;
-	const TEST_ROUNDS = options.rounds || 1;
+	const testRounds = options.rounds || 1;
 
-	// The default value doesn't work because commander provides an array.
-	if ( branches.length === 0 ) {
-		branches = [ 'trunk' ];
-	}
+	// @ts-ignore
+	let localRef;
 
-	log(
-		formats.title( '\n💃 Performance Tests 🕺\n' ),
-		'\nWelcome! This tool runs the performance tests on multiple branches and displays a comparison table.\n' +
-			'In order to run the tests, the tool is going to load a WordPress environment on ports 8888 and 8889.\n' +
-			'Make sure these ports are not used before continuing.\n'
-	);
+	if ( runningInCI ) {
+		localRef = process.env.GITHUB_HEAD_REF;
+		// @ts-ignore
+	} else if ( await simpleGit( CWD ).checkIsRepo() ) {
+		// @ts-ignore
+		localRef = await simpleGit( CWD ).revparse( {
+			'--abbrev-ref': null,
+			HEAD: null,
+		} );
+	}
 
-	if ( ! runningInCI ) {
-		await askForConfirmation( 'Ready to go? ' );
+	if ( ! localRef ) {
+		throw new Error( 'Must be running from a Gutenberg repository root.' );
```

Comment:
I guess this my main issue with this PR. My hope for this tool was to be plugin independent, I wanted to transform it into a package that any plugin can use to run perf tests, or any other command.

Reply:
OK, that should not be too difficult, as the logic for fetching and building from the origin is there. After checking [your other comment above](https://github.com/WordPress/gutenberg/pull/49238#discussion_r1167095353), I'm curious whether we should account for the fact that the script can run from the Gutenberg root. IMO it would make a lot of sense, as it would not break the original purpose to be an independent tool while being especially helpful in these situations:
1. **When debugging locally**, as it enables working on both the CLI perf tool and perf tests.
2. **When running from CI**, because the branch we run the workflow from is always a primary reference, enabling us to speed things up by reusing it and its dependencies (restored from the cache).